### ISSUE TITLE
Pull single side functions out of view

### ISSUE THREAD

**Issue by jordansread** [USGS-R/gsplot#290]
This is touched in discussion for #262 , but I'd like to make it a formal proposal w/ this issue. 

Here is what I am thinking:
`views` should only contain things that require the geometry of 2 (or more) sides to be defined
`sides` (a new top-level addition to the object) would contain things that are only relevant to a single side, such as `axis` calls, `mtext` calls, etc
`par` and `legend` stay as-is, as top-level. 

So an object like this:

``` r
gs <- gsplot()  %>% 
     points(1:20, 3:22, col="blue", legend.name="Points 1") 
```

would have this:

``` r
names(gs)
[1] "view.1.2" "par"      "legend"      "side.1"      "side.2"
```

These kinds of things: https://github.com/USGS-R/gsplot/blob/master/R/print.R#L83 would then be handed in the `sides` functions

**Comment by jordansread** [comment]
Likewise, 

``` r
gs <- gsplot()  %>% 
     points(1:20, 3:22, col="blue", legend.name="Points 1") %>% 
     points(20:1, 3:22, side=c(3,2), col="blue", legend.name="Points 1")
names(gs)
[1] "view.1.2" "view.3.2" "par"      "legend"        "side.1"      "side.2"      "side.3"
```

**Comment by lindsayplatt** [comment]
I like this idea, but if you aren't explicitly making a call to a function that only impacts one side (axis, mtext, etc), would the top-level single sides need to be there?

**Comment by jordansread** [comment]
I think they would be there no matter what. This way we can store some extra info in there, like when a user specifies xlim for a side, there can be a slot for it. Would be `NULL` by default

**Comment by jordansread** [comment]
plus, if you don't call axis, the `side.1` is where the default axis call would live.

**Comment by jiwalker-usgs** [comment]
I'd also like to add to this that we try to not have duplicate data as much as possible.  So rather than view.1.2 having `xlim` and `ylim`, side.1 has `lim` and side.2 has `lim` and view.1.2 knows to go there for `xlim` and `ylim`.

**Comment by jordansread** [comment]
yes, exactly. Fixes the overplotting issue too

**Comment by lindsayplatt** [comment]
That axis overplotting thing is really frustrating to look at. My plots look _almost_ perfect

**Comment by jordansread** [comment]
so close

**Comment by jordansread** [comment]
fixed w #313
